tests/php/Dev/Tasks/LegacyThumbnailMigrationHelperTest.php

```diff
@@ -100,7 +126,10 @@ public function testMigratesThumbnailsForProtectedFiles()
                 new ParsedFileID($image->Filename, $image->Hash),
                 $store->getProtectedFilesystem()
             );
-        $this->assertCount(1, $protectedVariants);
+
+        // findVariants() returns *both* the original file and the variant
+        // See FileIDHelperResolutionStrategyTest->testFindVariant()
+        $this->assertCount(2, $protectedVariants);
```

Comment:
`findVariants` returns a generator. Technically, those are not countable. PHPUnit seems to get around that by iterating over the generator until it reaches the end or goes over the limit. That does meant that when the test fails, you always get a "Failed asserting that actual size 0 matches expected size 2." even if the actual size is something other than zero.

I'm not going any where with this ... just pointing it out. I was half way going down the rabbit hole when I realised that it wasn't a problem.

Reply:
Actually, it is broken.

The moved variant is on the public store. You are looking on the protected store, so you are not getting the variant at all.

You are getting the main file twice:
* Once because it matches the NaturalFileIDHelper
* Once because it matches the LegacyFileIDHelper

Reply:
Aaaah great catch! This is a super gnarly test. I've removed it with fa4b14c013b2b71c6984ac03466f8c1c3c599c5b, and added it as a known limitation. See comment on main thread for more rationale: https://github.com/silverstripe/silverstripe-assets/pull/242#issuecomment-488155213